store/s3.go

```diff
@@ -11,41 +11,63 @@ import (
 	"github.com/minio/minio-go/v7/pkg/credentials"
 )
 
+const (
+	S3CredentialStatic  S3CredentialType = "static"
+	S3CredentialIAM     S3CredentialType = "iam"
+	S3CredentialUnknown S3CredentialType = "unknown"
+)
+
+type S3CredentialType string
 type S3Config struct {
-	Bucket          string
-	Endpoint        string
-	AccessKeyID     string
-	AccessKeySecret string
-	Profiling bool
+	S3CredentialType S3CredentialType
+	Bucket           string
+	Path             string
+	Endpoint         string
+	AccessKeyID      string
+	AccessKeySecret  string
+	Profiling        bool
 }
 
 type S3Store struct {
 	cfg    S3Config
 	client *minio.Client
-	stats *Stats
+	stats  *Stats
 }
 
 func NewS3(cfg S3Config) (*S3Store, error) {
```

Comment:
knit - this function could be refactored to support less duplication if we were to add a receiver to S3 config for generating the client credentials `func (cfg *Config) Credentials() minio.credentials` and only have a single instillation of the client in the s3 constructor